omniqueue/src/backends/redis/fallback.rs

```diff
@@ -61,29 +64,52 @@ async fn receive_with_timeout<R: RedisConnection>(
         .await
         .map_err(QueueError::generic)?;
 
-    key.map(|key| make_delivery(consumer, &key)).transpose()
+    match payload {
+        Some(old_payload) => {
+            let (payload, num_receives) = internal_from_list(&old_payload)?;
+            Some(internal_to_delivery(
+                (payload.to_vec(), num_receives),
+                consumer,
+                old_payload,
+            ))
+            .transpose()
+        }
+        None => Ok(None),
+    }
 }
 
-fn make_delivery<R: RedisConnection>(consumer: &RedisConsumer<R>, key: &[u8]) -> Result<Delivery> {
-    let (_, payload) = from_key(key)?;
-
+fn internal_to_delivery<R: RedisConnection>(
+    internal: InternalPayloadOwned,
+    consumer: &RedisConsumer<R>,
+    old_payload: Vec<u8>,
+) -> Result<Delivery> {
+    let (payload, num_receives) = internal;
     Ok(Delivery::new(
-        payload.to_owned(),
+        payload,
         RedisFallbackAcker {
             redis: consumer.redis.clone(),
             processing_queue_key: consumer.processing_queue_key.clone(),
-            key: key.to_owned(),
+            old_payload,
             already_acked_or_nacked: false,
+            max_receives: consumer.max_receives,
+            num_receives,
         },
     ))
 }
 
-struct RedisFallbackAcker<M: ManageConnection> {
-    redis: bb8::Pool<M>,
-    processing_queue_key: String,
-    key: RawPayload,
+pub(super) struct RedisFallbackAcker<M: ManageConnection> {
+    pub(super) redis: bb8::Pool<M>,
+    pub(super) processing_queue_key: String,
+    // We delete based on the payload -- and since the
```

Comment:
Why all the `pub(super) visibilities? I'm not seeing any new code that accesses this stuff in the parent module.